src/main/java/bio/terra/service/filedata/flight/ingest/BucketNameStep.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.service.filedata.flight.FileMapKeys;
+import bio.terra.service.profile.flight.ProfileMapKeys;
+import bio.terra.service.resourcemanagement.DataLocationSelector;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import java.util.UUID;
+
+public class BucketNameStep implements Step {
+
+    private final UUID datasetId;
+    private final DataLocationSelector dataLocationSelector;
+
+    public BucketNameStep(UUID datasetId, DataLocationSelector dataLocationSelector) {
+        this.datasetId = datasetId;
+        this.dataLocationSelector = dataLocationSelector;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext flightContext) {
+        FlightMap workingMap = flightContext.getWorkingMap();
+        bio.terra.model.BillingProfileModel billingProfile =
```

Comment:
Could you import `bio.terra.model.BillingProfileModel` for consistency?

Reply:
Sigh yeah, I haven't figured out how to get intellij to do this for the generated classes

Reply:
I recently did `rm -r .idea` and recreated the project from scratch. It seems to be working better now.